### PR TITLE
feat: Implement Response class and tests

### PR DESCRIPTION
**Pull request by ahume**
First pass.

### CODE REVIEW

File: src/utils/http.ts [hunk 1]
```diff
@@ -2,8 +2,15 @@
...
 const methods = ["DELETE", "GET", "HEAD", "OPTIONS", "POST", "PUT"];
+const redirectStatuses = [301, 302, 303, 307, 308];
 
 export function normalizeMethod(method): string {
   let upcased = method.toUpperCase();
   return (methods.indexOf(upcased) > -1) ? upcased : method;
 }
+
+export function isValidRedirectStatus(status): Boolean {
+  return redirectStatuses.some((v) => {
```
Comment (phuu) [inline src/utils/http.ts]: v => v === status

File: src/Response/Response.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+export default class Response extends Body implements FetchResponse {
+  type: string;
+  url: string;
+  status: number;
+  ok: boolean;
+  statusText: string;
+  headers: ResponseHeaders;
+
+  constructor(body, init: FetchResponseInit = {}) {
```
Comment (phuu) [inline src/Response/Response.ts]: `body = ""`

Then you shouldn't have to supply any args in the `fetch` method.

**Comment by phuu** [comment]
Looking really good!

File: src/utils/http.ts (hunk 1 above)
Comment (ahume) [inline src/utils/http.ts]: Done.

File: src/Response/Response.ts (hunk 2 above)
Comment (ahume) [inline src/Response/Response.ts]: Done.

**Pull request merged**